### ISSUE TITLE
gradle 3.5: "Resolving configuration 'apiElements' directly is not allowed"

### ISSUE THREAD

**Issue by pcarrier** [stempler/gradle-versioneye-plugin#36]
After upgrading from 3.4.x to 3.5.x, `versioneyeUpdate` fails with:

```
$ ./gradlew versioneyeUpdate --debug --stacktrace
[...]
16:09:22.911 [INFO] [org.gradle.execution.taskgraph.AbstractTaskPlanExecutor] :versioneye-pom (Thread[Task worker,5,main]) started.
16:09:22.911 [LIFECYCLE] [class org.gradle.internal.buildevents.TaskExecutionLogger] :versioneye-pom
16:09:22.911 [DEBUG] [org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter] Starting to execute task ':versioneye-pom'
16:09:22.911 [INFO] [org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter] Caching disabled for task ':versioneye-pom': Caching has not been enabled for the task
16:09:22.911 [INFO] [org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter] Putting task artifact state for task ':versioneye-pom' into context took 0.0 secs.
16:09:22.911 [DEBUG] [org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter] Determining if task ':versioneye-pom' is up-to-date
16:09:22.911 [INFO] [org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter] Executing task ':versioneye-pom' (up-to-date check took 0.0 secs) due to:
  Task has not declared any outputs.
16:09:22.911 [DEBUG] [org.gradle.api.internal.tasks.execution.SkipCachedTaskExecuter] Determining if task ':versioneye-pom' is cached already
16:09:22.911 [DEBUG] [org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter] Executing actions for task ':versioneye-pom'.
16:09:22.912 [DEBUG] [org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter] Removed task artifact state for {} from context.
16:09:22.912 [DEBUG] [org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter] Finished executing task ':versioneye-pom'
16:09:22.912 [LIFECYCLE] [class org.gradle.internal.buildevents.TaskExecutionLogger] :versioneye-pom FAILED
16:09:22.912 [INFO] [org.gradle.execution.taskgraph.AbstractTaskPlanExecutor] :versioneye-pom (Thread[Task worker,5,main]) completed. Took 0.001 secs.
16:09:22.912 [DEBUG] [org.gradle.execution.taskgraph.AbstractTaskPlanExecutor] Task worker [Thread[Task worker Thread 2,5,main]] finished, busy: 0.0 secs, idle: 0.001 secs
16:09:22.912 [DEBUG] [org.gradle.internal.operations.DefaultBuildOperationWorkerRegistry] Worker root.1 completed (0 in use)
16:09:22.912 [DEBUG] [org.gradle.execution.taskgraph.AbstractTaskPlanExecutor] Task worker [Thread[Task worker,5,main]] finished, busy: 0.001 secs, idle: 0.0 secs
16:09:22.912 [DEBUG] [org.gradle.execution.taskgraph.AbstractTaskPlanExecutor] Task worker [Thread[Daemon worker Thread 19,5,main]] finished, busy: 0.0 secs, idle: 0.001 secs
16:09:22.912 [DEBUG] [org.gradle.execution.taskgraph.AbstractTaskPlanExecutor] Task worker [Thread[Task worker Thread 4,5,main]] finished, busy: 0.0 secs, idle: 0.001 secs
16:09:22.912 [DEBUG] [org.gradle.execution.taskgraph.AbstractTaskPlanExecutor] Task worker [Thread[Task worker Thread 5,5,main]] finished, busy: 0.0 secs, idle: 0.001 secs
16:09:22.912 [DEBUG] [org.gradle.execution.taskgraph.AbstractTaskPlanExecutor] Task worker [Thread[Task worker Thread 6,5,main]] finished, busy: 0.0 secs, idle: 0.001 secs
16:09:22.912 [DEBUG] [org.gradle.execution.taskgraph.AbstractTaskPlanExecutor] Task worker [Thread[Task worker Thread 3,5,main]] finished, busy: 0.0 secs, idle: 0.001 secs
16:09:22.912 [DEBUG] [org.gradle.execution.taskgraph.AbstractTaskPlanExecutor] Task worker [Thread[Task worker Thread 7,5,main]] finished, busy: 0.0 secs, idle: 0.001 secs
16:09:22.913 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]
16:09:22.914 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] FAILURE: Build failed with an exception.
16:09:22.914 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]
16:09:22.914 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * What went wrong:
16:09:22.914 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] Execution failed for task ':versioneye-pom'.
16:09:22.914 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] > Resolving configuration 'apiElements' directly is not allowed
16:09:22.914 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]
16:09:22.914 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * Exception is:
16:09:22.914 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':versioneye-pom'.
16:09:22.914 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:98)
16:09:22.914 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:68)
16:09:22.914 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.tasks.execution.SkipCachedTaskExecuter.execute(SkipCachedTaskExecuter.java:94)
16:09:22.914 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:62)
16:09:22.914 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.tasks.execution.ResolveBuildCacheKeyExecuter.execute(ResolveBuildCacheKeyExecuter.java:51)
16:09:22.914 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
16:09:22.914 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:88)
16:09:22.914 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:46)
16:09:22.914 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:51)
16:09:22.914 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
16:09:22.914 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:47)
16:09:22.914 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
16:09:22.914 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
16:09:22.914 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.execute(DefaultTaskGraphExecuter.java:236)
16:09:22.915 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.execute(DefaultTaskGraphExecuter.java:228)
16:09:22.915 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.Transformers$4.transform(Transformers.java:169)
16:09:22.915 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:106)
16:09:22.915 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:61)
16:09:22.915 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:228)
16:09:22.915 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:215)
16:09:22.915 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.processTask(AbstractTaskPlanExecutor.java:77)
16:09:22.915 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.run(AbstractTaskPlanExecutor.java:58)
16:09:22.915 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
16:09:22.915 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.concurrent.StoppableExecutorImpl$1.run(StoppableExecutorImpl.java:46)
16:09:22.915 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] Caused by: java.lang.IllegalStateException: Resolving configuration 'apiElements' directly is not allowed
16:09:22.915 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.assertResolvingAllowed(DefaultConfiguration.java:825)
16:09:22.915 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.resolveToStateOrLater(DefaultConfiguration.java:428)
16:09:22.915 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.getResolvedConfiguration(DefaultConfiguration.java:423)
16:09:22.915 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration_Decorated.getResolvedConfiguration(Unknown Source)
16:09:22.915 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.metaobject.BeanDynamicObject$MetaClassAdapter.getProperty(BeanDynamicObject.java:213)
16:09:22.915 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.metaobject.BeanDynamicObject.getProperty(BeanDynamicObject.java:156)
16:09:22.915 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.metaobject.CompositeDynamicObject.getProperty(CompositeDynamicObject.java:55)
16:09:22.915 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.metaobject.AbstractDynamicObject.getProperty(AbstractDynamicObject.java:60)
16:09:22.915 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration_Decorated.getProperty(Unknown Source)
16:09:22.915 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.standardout.gradle.plugin.versioneye.PomTask$_create_closure1_closure6.doCall(PomTask.groovy:79)
16:09:22.915 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.standardout.gradle.plugin.versioneye.PomTask$_create_closure1.doCall(PomTask.groovy:76)
16:09:22.915 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.standardout.gradle.plugin.versioneye.PomTask.create(PomTask.groovy:75)
16:09:22.915 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
16:09:22.916 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.doExecute(DefaultTaskClassInfoStore.java:141)
16:09:22.916 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.execute(DefaultTaskClassInfoStore.java:134)
16:09:22.916 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.execute(DefaultTaskClassInfoStore.java:123)
16:09:22.916 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:692)
16:09:22.916 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:675)
16:09:22.916 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.execute(ExecuteActionsTaskExecuter.java:115)
16:09:22.916 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.execute(ExecuteActionsTaskExecuter.java:109)
16:09:22.916 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.Transformers$4.transform(Transformers.java:169)
16:09:22.916 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:106)
16:09:22.916 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:56)
16:09:22.916 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:109)
16:09:22.916 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:90)
16:09:22.916 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   ... 23 more
16:09:22.916 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]
16:09:22.916 [LIFECYCLE] [org.gradle.internal.buildevents.BuildResultLogger]
16:09:22.916 [LIFECYCLE] [org.gradle.internal.buildevents.BuildResultLogger] BUILD FAILED
16:09:22.916 [LIFECYCLE] [org.gradle.internal.buildevents.BuildResultLogger]
16:09:22.916 [LIFECYCLE] [org.gradle.internal.buildevents.BuildResultLogger] Total time: 1.203 secs
[...]
```

**Comment by stempler** [comment]
That's the same issue as #34 - it fails because they introduced unresolvable configurations in Gradle. See https://github.com/stempler/gradle-versioneye-plugin/issues/34#issuecomment-283512636 for a workaround.

**Comment by pcarrier** [comment]
@stempler I ran into #34 and introduced the workaround already. Build works with 3.4.x, fails with 3.5.x.

```
allprojects {
    versioneye {
        includeSubProjects = true
        exclude configurations.findAll { !it.canBeResolved }*.name as String[]
    }
}
```

**Comment by stempler** [comment]
So does the configuration `apiElements` then yield `true` for `canBeResolved` or is there another problem?

**Comment by pcarrier** [comment]
@stempler no idea.

**Comment by Kostronor** [comment]
@stempler Got the same problem, tried the workaround in #34 and it still fails.

Example: https://github.com/Minecolonies/minecolonies/pull/871

**Comment by stempler** [comment]
Can you try again with the [current snapshot](https://github.com/stempler/gradle-versioneye-plugin#using-the-current-snapshot)?
The reason why it fails even if the workaround is used is most likely that subprojects are included and there are configurations that are not resolvable that don't occur in the main project.

**Comment by benjamin-olx** [comment]
@stempler Can confirm, using latest snapshot fixes it.

**Comment by sbraconnier** [comment]
@stempler I also had this problem with my multi-modules project using the latest release and I also confirm that the current snapshot fixes it. Thanks.

**Comment by stempler** [comment]
Fix contained in 1.5.0 release.
